### DISCUSSION TITLE
Preview του κώδικα

### DISCUSSION THREAD

**Discussion by mar-stamp** [merkourisa/Augmented-Reality#9] [Q&A]
Έχω φτιάξει τον κώδικά μου αλλά πως μπορώ να κάνω ένα preview για να σιγουρευτώ ότι δουλεύει;

**Comment by merkourisa** [comment]
@mar-stamp Από την στιγμή που έχεις κατεβάσει τον κώδικα από το github στον υπολογιστή σου και τον έχει επεξεργαστεί με ένα IDE σύμφωνα με το ζητούμενα του 1ου παραδοτέου για να το τεστάρεις θα πρέπει να ρυθμίσεις και να τρέξεις έναν τοπικό web server ([python 3 http.server](https://developer.mozilla.org/en-US/docs/Learn/Common_questions/set_up_a_local_testing_server)).

Δες αυτό το [βίντεο](https://www.youtube.com/watch?v=-Gu0mUjR9Vo) και αυτό το [βίντεο](https://www.youtube.com/watch?v=lE6Y6M9xPLw&t) που εξηγούν πως τρέχουμε τον python local web server.

Τα βήματα είναι τα εξής :
1.  Κάνεις εγκατάσταση την [python](https://www.python.org/).
2. Ανοίγεις την Γραμμή Εντολών (Command Line) ή το Windows PowerShell.
3. Πληκτρολογείς **python -V** και σου επιστρέφει την εγκατεστημένη έκδοση της python (Python 3.10.2 αν έχει εγκαταστήσει την τελευταία έκδοση).
4. Μέσω της γραμμής εντολών μεταβαίνεις στο φάκελο που έχεις τον κώδικά σου πχ cd Documents\Augmented_Reality\marker_based 
5. Eκτέλεις στη γραμμή εντολών την εντολή **python -m http.server** για να ξεκινήσεις τον τοπικό Web server.
6. Ανοίγεις τον chrome ή οποιοδήποτε άλλο browser και στη διεύθυνση δίνεις **http://localhost:8000/**
7. Θα δεις την εφαρμογή σου να τρέχει _τοπικά_ στον browser.

Εναλλακτικά θα μπορούσες να χρησιμοποιήσεις ένα online IDE όπως το [codepen](https://codepen.io/pen/) αλλά θα πρότεινα τον τοπικό Web server  γιατί στα επόμενα παραδοτέα θα χρειαστεί να διαβάσετε αρχεία μέσα από τον φάκελο /assets και θα υπάρξει δυσκολία με το codepen.

PS. Μέχρι την Παρασκευή θα έχει εγκατασταθεί η python στο εργαστήριο της πληροφορικής (αριστερή αίθουσα)

**Comment by ievagelatoumanolia** [comment]
@merkourisa Καλησπέρα, πατάω το http://localhost:8000/ που αναφέρετε παραπάνω και μου λέει ότι δεν είναι δυνατή η πρόσβαση σε αυτό τον ιστότοπο. Τι μπορεί να συμβαίνει;

**Comment by merkourisa** [comment]
@ievagelatoumanolia καλησπέρα, έχεις εκτελέσει τα [παραπάνω βήματα](https://github.com/merkourisa/Augmented-Reality/discussions/9#discussioncomment-2349464);

Στο 5ο βήμα όταν δίνεις την εντολή python -m http.server θα πρέπει να ξεκινάει ο τοπικός Web server.
